core/src/test/java/org/apache/iceberg/rest/requests/TestCommitTransactionRequestParser.java

```diff
@@ -0,0 +1,245 @@
+ */
+package org.apache.iceberg.rest.requests;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.iceberg.MetadataUpdate;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.rest.requests.UpdateTableRequest.UpdateRequirement;
+import org.junit.jupiter.api.Test;
+
+public class TestCommitTransactionRequestParser {
+
+  @Test
+  public void nullAndEmptyCheck() {
+    assertThatThrownBy(() -> CommitTransactionRequestParser.toJson(null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid commit tx request: null");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson((JsonNode) null))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse commit tx request from null object");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson("{}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: table-changes");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson("{\"table-changes\":{}}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse commit tx request from non-array: {}");
+
+    assertThatThrownBy(() -> CommitTransactionRequestParser.fromJson("{\"table-changes\":[]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid table changes: empty");
+  }
+
+  @Test
+  public void invalidTableIdentifier() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"ns1.table1\" : \"ns1.table1\"}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: identifier");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\" : {}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing string: name");
+
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\" : { \"name\": 23}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse to a string value: name: 23");
+  }
+
+  @Test
+  public void invalidRequirements() {
+    assertThatThrownBy(
+            () ->
+                CommitTransactionRequestParser.fromJson(
+                    "{\"table-changes\":[{\"identifier\":{\"namespace\":[\"ns1\"],\"name\":\"table1\"}}]}"))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot parse missing field: requirements");
```

Comment:
I think this should be allowed. We don't want to fail a request because the client was configured to omit empty fields.

Reply:
good point, I'll make sure that empty `updates` and `requirements` don't fail the parsing code